nixos/modules/services/continuous-integration/gitea-actions-runner.nix

```diff
@@ -203,6 +203,8 @@ in
             TOKEN = "${instance.token}";
           } // optionalAttrs (wantsPodman) {
             DOCKER_HOST = "unix:///run/podman/podman.sock";
+          } // {
+            HOME = "/var/lib/gitea-runner/${name}";
```

Comment:
Are there negative implications for this change? For example will a local runner create files in this directory?

Reply:
On my `:host` runner I see an additional `.cache` directory, which holds `act` and `actcache`.

`.cache/act/actions-checkout@v4` for example.

We could probably use `/var/cache/` instead /shrug

I can't think of any negative implications either way though.

Reply:
Yeah, I'm not too worries about the act directories as long as other state isn't getting stored there. Adding `CacheDirectory` to the serviceConfig and pointing to it seems reasonable, but isn't a huge deal.

Reply:
Without the addition of `HOME`, my forgejo-runner breaks, with particularly hard to debug error messages.


when using
`echo "all good"`
```
bash: .cache/act/1a25cf138c689101/act/workflow/0.sh: No such file or directory
```

When using `actions/checkout`
```
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] setupEnv => map[ACT:true ACTIONS_CACHE_URL:http://10.0.0.5:43687/ ACTIONS_RESULTS_URL:http://codex.cypress.local ACTIONS_RU>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] evaluating expression ''
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] expression '' evaluated to 'false'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] Skipping step 'cat testfile' due to ''
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] setupEnv => map[ACT:true ACTIONS_CACHE_URL:http://10.0.0.5:43687/ ACTIONS_RESULTS_URL:http://codex.cypress.local ACTIONS_RU>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] evaluating expression 'always()'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] expression 'always()' evaluated to 'true'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] ⭐ Run Post http://codex.cypress.local/actions/checkout@main
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] run post step for 'http://codex.cypress.local/actions/checkout@main'
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] [DEBUG] executing remote job container: [node .cache/act/28cb059d3c646cfe/act/actions/http---codex.cypress.local-actions-checkout@m>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | node:internal/modules/cjs/loader:1137
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   throw err;
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   ^
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | Error: Cannot find module '/var/lib/private/gitea-runner/host/.cache/act/28cb059d3c646cfe/hostexecutor/.cache/act/28cb059d3c646>
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at Module._load (node:internal/modules/cjs/loader:975:27)
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |     at node:internal/main/run_main_module:28:49 {
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   code: 'MODULE_NOT_FOUND',
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |   requireStack: []
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | }
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   |
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   | Node.js v18.19.1
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat]   ❌  Failure - Post http://codex.cypress.local/actions/checkout@main
Apr 24 09:24:54 cypress act_runner[570319]: [/checkout-cat] Cleaning up container for job checkout-cat
```

If you're running this on 23.11
```
systemd.services.gitea-runner-<instance>.environment.HOME = "/var/lib/gitea-runner/<instance>";
```
will clear up any strange errors related to this issue.